GSA-TTS/FAC
# PDF Upload Frontend

# PDF Upload Frontend - Django View/HTML/JS

Issue https://github.com/GSA-TTS/FAC/issues/941

## Changes
1. New `UploadReportView` under `audit` that handles the getting/posting of audit report PDF form. 
2. New HTML `upload-report.html`, which is a formatted form that should match very closely to the associated [Figma doc](https://www.figma.com/file/1bNncVt1TcFFDhQslqFC2o/FAC-Playground?node-id=5077%3A29727&mode=dev). 
3. New JS file to handle enabling/disabling of the submit button.

Form validation is handled in Django since it's simpler than other file uploads. Submitting the form will post to the Django view, and if errors are found the page is re-rendered with those errors. This means we can set custom errors when a user uploads a PDF that fails validation (locked, too big, etc.)

## How to test
Switch to this branch and run normally. These changes should not require a rebuild, but may require you to restart your container. Navigate to your [audit home page](http://localhost:8000/audit/submission-progress/YOURIDHERE), and click "[Upload the audit report PDF](http://localhost:8000/report_submission/upload-report/YOURIDHERE)". Fill out the form. Successful entries will redirect you to your audit home page, and will not currently save anything to the DB. 

You shouldn't be able to enter or select any negative values for the page numbers, and you should only be able to enter PDFs. Once all required field are filled in, the submit button will enable.

Screenshots:
![image](https://github.com/GSA-TTS/FAC/assets/91098850/e798f7ca-596c-4fc2-8a2a-74df823cb7cf)
![image](https://github.com/GSA-TTS/FAC/assets/91098850/0b9f4a0a-ab1f-4946-8531-7632880f307a)
